Update links in readme and contributing. (#6948)
geodynamics/aspect: README.md

 - The current [Online Documentation](https://aspect-documentation.readthedocs.io/en/latest/)
 - [How to cite ASPECT](https://aspect.geodynamics.org/cite.html)
 - For questions on the source code of ASPECT, portability, installation, new or existing features, etc., use the [ASPECT forum](https://community.geodynamics.org/c/aspect). This forum is where the ASPECT users and developers all hang out.

>  - ASPECT is primarily based on the deal.II library. If you have particular questions about deal.II, contact the [deal.II discussion groups](https://dealii.org/getting_help/).

Reviewer: ```suggestion
 - ASPECT is primarily based on the deal.II library. If you have particular questions about deal.II, have a look at the [deal.II help pages](https://dealii.org/getting_help/).
```
